components/IdleTimeout.tsx

```diff
@@ -0,0 +1,139 @@
+  })
+  const t = locale === 'en' ? en : fr
+  const id = useId()
+
+  const handleOnIdle = () => {
+    router.push('/auth/logout')
+  }
+
+  const handleOnIdleContinueSession = () => {
+    setModalOpen(false)
+    reset()
+  }
+
+  const { reset, getRemainingTime } = useIdleTimer({
+    onIdle: handleOnIdle,
+    onPrompt: () => setModalOpen(true),
+    promptBeforeIdle: promptBeforeIdle ?? 5 * 60 * 1000, //5 minutes
+    timeout: timeout ?? 15 * 60 * 1000, //15 minutes
+  })
+
+  const tick = useCallback(() => {
+    const minutes = Math.floor(getRemainingTime() / 60000)
+    const seconds = Math.floor((getRemainingTime() / 1000) % 60).toFixed(0)
+    setTimeRemaining({ seconds, minutes })
+  }, [getRemainingTime])
+
+  useEffect(() => {
+    setInterval(tick, 1000)
+  }, [tick])
+
+  return (
+    <Modal
+      className="flex h-full justify-center bg-black/75"
+      isOpen={modalOpen}
+      onRequestClose={handleOnIdleContinueSession}
+      contentLabel={t.bannerHeading}
+      id={`${id}-timeout-modal`}
+    >
+      <FocusOn enabled={modalOpen}>
+        <div
+          className="m-8 h-fit rounded bg-white p-4 sm:mx-24 sm:mt-24 md:p-16"
+          data-cy="timeOutModal"
+          data-testid="modal"
+          aria-describedby={`${id}-timeout-modal-header`}
+        >
+          <div className="flex justify-between pb-5">
+            <div
+              className="font-display text-3xl font-bold"
+              role="heading"
+              aria-level={1}
+              id={`${id}-timeout-modal-header`}
+            >
+              {t.bannerHeading}
+            </div>
+            <button
+              data-cy="x-button"
+              type="button"
+              aria-label={'Close Modal'}
+              onClick={handleOnIdleContinueSession}
+              data-gc-analytics-customclick={`ESDC-EDSC:${refPageAA}:Close-Fermer`}
+            >
+              <FontAwesomeIcon
+                aria-hidden="true"
+                icon={icon['xmark']}
+                size="xl"
+              />
+            </button>
+          </div>
+
+          <div className="flex">
+            <div className="items-top flex flex-none justify-center pr-3">
+              <FontAwesomeIcon
+                icon={icon['triangle-exclamation']}
+                size="lg"
+                color="orange"
+              />
+            </div>
+            <div className="flex-auto" id={`${id}-timeout-modal-desc`}>
+              <p className="mr-6">{t.bannerContent.notActive}</p>
+              <p className="mr-6 font-bold">
+                {t.bannerContent.signOut} {timeRemaining.minutes}
+                {t.bannerMinutesAnd} {timeRemaining.seconds} {t.bannerSeconds}.
+              </p>
+            </div>
+          </div>
+
+          <div className="flex pt-10 md:space-x-12">
+            <Button
+              id="sign-out"
+              text={t.signOutLinkText}
+              onClick={handleOnIdle}
+              style="secondary"
```

Comment:
can you pass the AA ref? Right now it's showing as undefined.

Reply:
:+1: